doc/source/visualization.rst

```diff
@@ -456,28 +456,35 @@ columns:
 
 .. _visualization.box.return:
 
-Basically, plot functions return :class:`matplotlib Axes <matplotlib.axes.Axes>` as a return value.
-In ``boxplot``, the return type can be changed by argument ``return_type``, and whether the subplots is enabled (``subplots=True`` in ``plot`` or ``by`` is specified in ``boxplot``).
+.. warning::
+
+   The default changed from ``'dict'`` to ``'axes'`` in version 0.19.0.
+
+Most plot functions return a scalar or an array of :class:`matplotlib Axes <matplotlib.axes.Axes>`.
+In ``boxplot``, the return type can be controlled by the ``return_type``, keyword. The valid choices are ``{"axes", "dict", "both"}``. If the ``by`` argument is ``None``,
 
-When ``subplots=False`` / ``by`` is ``None``:
+* ``'axes'`` returns a single matplotlib axes.
+* ``'dict'`` returns a dict of matplotlib artists, similar to the matplotlib boxplot function.
+* ``'both'`` returns a named tuple of axes and dicts.
 
-* if ``return_type`` is ``'dict'``, a dictionary containing the :class:`matplotlib Lines <matplotlib.lines.Line2D>` is returned. The keys are "boxes", "caps", "fliers", "medians", and "whiskers".
-   This is the default of ``boxplot`` in historical reason.
-   Note that ``plot.box()`` returns ``Axes`` by default same as other plots.
-* if ``return_type`` is ``'axes'``, a :class:`matplotlib Axes <matplotlib.axes.Axes>` containing the boxplot is returned.
-* if ``return_type`` is ``'both'`` a namedtuple containing the :class:`matplotlib Axes <matplotlib.axes.Axes>`
-   and :class:`matplotlib Lines <matplotlib.lines.Line2D>` is returned
+Faceted boxplots, created by ``Groupby.boxplot`` or ``DataFrame.boxplot`` with the ``by``
+keyword will return either a NumPy array of ``axes``, when ``return_type='axes'``,
+or a ``Series`` of ``return_type``. Each column of the input ``DataFrame``
+will get a single facet.
```

Comment:
This is also not true in case of `groupby.boxplot` (there the facets are the groups)

Given those difference, maybe only speak about `DataFrame.boxplot` here?